lib/compareValues.js

```diff
@@ -78,7 +80,9 @@ function compareValues (a, b) {
     case 'NULL':
       return spaceship(a, b)
     case 'PTYPE<TIME>':
-      return spaceship(a.getTime(), b.getTime())
+      const timeA = _.isDate(a) ? a.getTime() : a.epoch_time * 1000
```

Comment:
It does, and yeah, it could be a bad idea (because datetimes seem to require a lot of space and PKs are limited to 127 characters). Still, this may easily happen if someone wants timestamps as PKs (whole or a part of a composite key) and doesn't want to convert native objects to strings or numbers.